web/Controllers/LandlordController.cs

```diff
@@ -44,25 +48,37 @@ public ActionResult RegisterGet()
     public async Task<ActionResult> RegisterPost([FromForm] LandlordProfileModel createModel)
     {
         // This does checks based on the annotations (e.g. [Required]) on LandlordProfileModel
-        if (!ModelState.IsValid)
-        {
-            return View("Register");
-        }
+        if (!ModelState.IsValid) return View("Register");
 
         var user = GetCurrentUser();
+        ILandlordService.LandlordRegistrationResult result;
+        int? id = null;
+
+        if (createModel.Unassigned && user.IsAdmin)
+        {
+            var resultTuple = await _landlordService.RegisterLandlord(createModel);
+            result = resultTuple.result;
+            id = resultTuple.id;

```

Comment:
Haven't tested it, but I think this works
```suggestion
            (result, id) = await _landlordService.RegisterLandlord(createModel);
```

Reply:
This seems to work :)